AtelierTomato.Markov/AtelierTomato.Markov.Renderer.Test/DiscordSentenceRendererTests.cs

```diff
@@ -0,0 +1,104 @@
+		[InlineData("> implying", "\\>implying")]
+		[InlineData("5 > 2", "5 \\> 2")]
+		public void RenderSimpleTextTest(string input, string output)
+		{
+			var target = new DiscordSentenceRenderer();
+
+			var result = target.Render(input);
+
+			result.Should().Be(output);
+		}
+
+
+		[Theory]
+		[InlineData("( this sentence is in parentheses )", "\\(this sentence is in parentheses\\)")]
+		[InlineData("i am god [ citation needed ] and you are stupid { you are DUMB }", "i am god \\[citation needed\\] and you are stupid \\{you are DUMB\\}")]
+		[InlineData("\" who are you quoting ? \"", "\\\"who are you quoting\\?\\\"")]
+		[InlineData("god \" jimmy \" jesus here with another minecraft video", "god \\\"jimmy\\\" jesus here with another minecraft video")]
+		public void RenderSurroundingCharactersTest(string input, string output)
+		{
+			var target = new DiscordSentenceRenderer();
+
+			var result = target.Render(input);
+
+			result.Should().Be(output);
+		}
+
+		[Theory]
+		[InlineData("i 've found a million dollars", "i\\'ve found a million dollars")]
+		[InlineData("what 's up gamers , I 'm out here gaming", "what\\'s up gamers\\, I\\'m out here gaming")]
+		[InlineData("the students ' council decided you die today", "the students\\' council decided you die today")]
+		[InlineData("and i can tell that doki doki literature club is shitty faux- anime normie garbage", "and i can tell that doki doki literature club is shitty faux\\-anime normie garbage")]
+		public void RenderContractionsTest(string input, string output)
+		{
+			var target = new DiscordSentenceRenderer();
+
+			var result = target.Render(input);
+
+			result.Should().Be(output);
+		}
+
+		[Theory]
+		[InlineData("e:ShihoLook:", "<:ShihoLook:402558230427074560>")]
+		[InlineData("e:ShihoLook: e:applesauce:", "<:ShihoLook:402558230427074560> applesauce")]
+		[InlineData("appe:ShihoLook:le", "app<:ShihoLook:402558230427074560>le")]
+		[InlineData("e:ShihoLook: e:ShihoLook: e:ShihoLook:", "<:ShihoLook:402558230427074560> <:ShihoLook:402558230427074560> <:ShihoLook:402558230427074560>")]
+		[InlineData("e:apple2:", "apple2")]
+		[InlineData("e:__:", "\\_\\_")]
+		public void RenderEmojisInCurrentTest(string input, string output)
+		{
+			var emote = new Emote(402558230427074560, "ShihoLook");
+
+			IEnumerable<Emote> currentEmojis = [emote];
+			IEnumerable<Emote> allEmojis = [];
+
+			var target = new DiscordSentenceRenderer();
+
+			var result = target.Render(input, currentEmojis, allEmojis);
+			result.Should().Be(output);
+		}
+
+		[Theory]
+		[InlineData("e:ShihoLook:", "<:ShihoLook:402558230427074560>")]
+		[InlineData("e:ShihoLook: e:applesauce:", "<:ShihoLook:402558230427074560> applesauce")]
+		[InlineData("appe:ShihoLook:le", "app<:ShihoLook:402558230427074560>le")]
+		[InlineData("e:ShihoLook: e:ShihoLook: e:ShihoLook:", "<:ShihoLook:402558230427074560> <:ShihoLook:402558230427074560> <:ShihoLook:402558230427074560>")]

```

Comment:
test case for animated emotes is missing, also try emotes with underscores, numbers, (are tildes valid? then those too)